src/global/reco/ReconstructedElectrons_factory.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2022, 2023 Daniel Brandenburg
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#pragma once
+
+// JANA
+#include <extensions/jana/JChainFactoryT.h>
+#include <JANA/JEvent.h>
+
+// algorithms
+#include <algorithms/reco/ElectronReconstruction.h>
+
+// services
+#include <services/log/Log_service.h>
+#include <extensions/spdlog/SpdlogExtensions.h>
+#include <extensions/spdlog/SpdlogMixin.h>
+
+namespace eicrecon {
+
+  class ReconstructedElectrons_factory :
+    public JChainFactoryT<edm4eic::ReconstructedParticle>,
+    public SpdlogMixin<ReconstructedElectrons_factory>
+  {
+
+    public:
+
+      explicit ReconstructedElectrons_factory(std::vector<std::string> default_input_tags) :
+        JChainFactoryT<edm4eic::ReconstructedParticle>(std::move(default_input_tags)) {
+        }
+
+      /** One time initialization **/
+      void Init() override {
+        // get plugin name and tag
+        auto app    = GetApplication();
+        auto plugin = GetPluginName();
+        auto prefix = plugin + ":" + GetTag();
+        InitDataTags(prefix);
+
+        // services
+        InitLogger(prefix, "info");
+        m_algo.init(m_log);
+      }
+
+      /** On run change preparations **/
+      void BeginRun(const std::shared_ptr<const JEvent> &event) override{}
+
+      /** Event by event processing **/
+      void Process(const std::shared_ptr<const JEvent> &event) override{
+        // Step 1. lets collect the Cluster associations from various detectors
+        std::vector<const edm4eic::MCRecoClusterParticleAssociationCollection*> in_clu_assoc;
+        for(auto& input_tag : GetInputTags()){
+          // only collect from the sources that provide ClusterAssociations
+          if ( input_tag.find( "ClusterAssociations" ) == std::string::npos ) {
+            continue;
+          }
+          m_log->trace( "Adding cluster associations from: {}", input_tag );
+          in_clu_assoc.push_back(
+            static_cast<const edm4eic::MCRecoClusterParticleAssociationCollection*>(event->GetCollectionBase(input_tag))
```

Comment:
Is this used elsewhere? I don't recall the need for static_cast.

Reply:
yes, see mattermost discussion, GetCollection<> is giving problems and this was suggested as a workaround